### DISCUSSION TITLE
vu meter with WS2812 how can i do that?

### DISCUSSION THREAD

**Discussion by bynauy** [tttapa/Control-Surface#1038] [Q&A]
i was trying an idea, put WS2812 as vu meter for each channel on my control surface, im following this example: 

```cpp
#include <Control_Surface.h>
#include <Adafruit_NeoPixel.h>

#define LED_PIN    6  // neopixel pin
#define LED_COUNT  8  // #neopixel

Adafruit_NeoPixel strip(LED_COUNT, LED_PIN, NEO_GRB + NEO_KHZ800);

USBMIDI_Interface midi;

// Instancia un objeto VULEDs con 12 LEDs.
MCU::VULEDs<12> vu {
  strip,                      // Neopixel
  1,                          // Número de pista [1, 8]
  MCU::VUDecay::Default,      // decay
};

// change mode
Button dotBarModeButton = A0;

void setup() {
  strip.begin();
  strip.show(); // Initialize all pixels to 'off'
  Control_Surface.begin();
  dotBarModeButton.begin();
}

void loop() {
  Control_Surface.loop();
  if (dotBarModeButton.update() == Button::Falling) //push
    vu.toggleMode(); // change mode
}
```
but, it doesnt work. the ws2812 didnt show the information, i check the conections and its all okay, if someone can helpme i would apreciate that.
thanks in advice.
PD sorry for my english

**Comment by tttapa** [comment]
> ```cpp
> // Instancia un objeto VULEDs con 12 LEDs.
> MCU::VULEDs<12> vu {
>   strip, 
> ```

I would be very surprised if this compiled. The first argument to the `VULEDs` class is a list of pin _numbers_, you cannot simply pass an unrelated type like the `Adafruit_NeoPixel` class.

---

As a starting point, I'd recommend trying out https://tttapa.github.io/Control-Surface-doc/Doxygen/d8/dd2/9_8Note-FastLED_8ino-example.html. Make sure you understand what's going on, and get the LEDs and MIDI to work in a simpler setting (by sending MIDI notes with the right values).

Then have a look at how this class drives the LEDs under the hood: https://tttapa.github.io/Control-Surface-doc/Doxygen/d0/d37/NoteCCKPRangeFastLED_8hpp_source.html#l00060

Next, look at the VU class and how it receives the MIDI messages for the VU meter: https://tttapa.github.io/Control-Surface-doc/Doxygen/db/dda/VU_8hpp_source.html#l00188

Finally, have a look at the VULEDs class, and create a copy of it and modify it to use the FastLED library instead of normal output pins. You can use the techniques used in the NoteCCKPRangeFastLED class: https://tttapa.github.io/Control-Surface-doc/Doxygen/de/d3d/VULEDs_8hpp_source.html#l00042
